terraform/03_gke_cluster.tf

```diff
@@ -95,7 +97,7 @@ resource "google_container_cluster" "gke" {
       }
     }
 
-    initial_node_count = 4
+    initial_node_count = 2
```

Comment:
Other than cost because you also increased the size of the nodes, any particular reason to have 2 nodes now? Could we have at least 3 nodes to have corum? Furthermore, with Online Boutique the recommendation is to have 4 nodes at least, especially with ASM: https://github.com/GoogleCloudPlatform/microservices-demo#quickstart-gke. So maybe let's keep 4?

Reply:
Yeah, we can switch from 2 n1-standard-4s back to 4 n1-standard-2s if that's preferred. I assumed all things equal, less nodes would have lower overhead. But, I'm happy to change if that's not the case

I am interested in if you have any advice to lower the costs though, as an 8 vCPU cluster can be expensive for a demo app. Is there a minimal ASM cluster recommendation? Could we get away comfortably with something like a e2-small?

Also, is the 8vCPU requirement a hard cap? Sandbox is used in Qwiklabs, and they require special approvals to run labs with more than 6 vCPUs

Reply:
The recommendation for ASM with the controlplane in-cluser, what you have now, is explained [here](https://cloud.google.com/service-mesh/docs/unified-install/anthos-service-mesh-prerequisites#cluster_requirements): `Anthos Service Mesh requires at least 8 vCPUs`. With that said, with ASM MCP (other comment) you have the advantage to remove the `istiod` component and its CPU/memory usage from in-cluster to outside the cluster. Very convenient to decrease size of the nodes of the cluster too.

For [Online Boutique](https://github.com/GoogleCloudPlatform/microservices-demo#quickstart-gke), the recommendation is `--machine-type=e2-standard-2 --num-nodes=4`.

The potential issue with 2 nodes is it's not respecting the quorum and best practice in case of failure while upgrading the clusters, etc. So 3 nodes at least could show best practice/recommendation here.